### PR TITLE
filter restricter fields

### LINKED ISSUE #10: Enhance protocol send additional fields to the restricter
Currently only the fqids of the found datasets are send to the
restricter. The restricter needs per collection a set of additional fields
which it should send in all cases (e.q. sequentials)

### PR DESCRIPTION
**Pull request by bastianjoel**
proposal to resolve #10
requires https://github.com/OpenSlides/openslides-autoupdate-service/pull/659

### CODE REVIEW

File: cmd/searchd/main.go [hunk 1]
```diff
@@ -89,7 +89,12 @@ func run(cfg *config.Config) error {
 
 	go authBackground(ctx, oserror.Handle)
 
-	return web.Run(ctx, cfg, authService, qs)
+	fields := make(map[string][]string)
+	for _, m := range searchModels.AsFilters() {
+		fields[m.Name] = append([]string{"id", "sequential_number"}, m.Items...)
+	}
```
Comment (bastianjoel) [inline cmd/searchd/main.go:95]: This will be improved with #9

File: pkg/web/http.go [hunk 2]
```diff
@@ -25,9 +25,10 @@ import (
 )
 
 type controller struct {
-	cfg  *config.Config
-	auth *auth.Auth
-	qs   *search.QueryServer
+	cfg    *config.Config
+	auth   *auth.Auth
+	qs     *search.QueryServer
+	fields *map[string][]string
```
Comment (s-l-teichmann) [inline pkg/web/http.go:31]: I see no reason here to pass a pointer to the map. Maps are already reference types.

**Review by s-l-teichmann — CHANGES_REQUESTED** [review]

File: pkg/web/http.go [hunk 3]
```diff
@@ -71,12 +72,24 @@ func (c *controller) search(w http.ResponseWriter, r *http.Request) {
 			}
 		*/
 
+		requestedFields := make(map[string][]string, 0)
```
Comment (s-l-teichmann) [inline pkg/web/http.go:75]: Drop the `0` or just write `requestedFields := map[string][]string{}`

**Review by s-l-teichmann — APPROVED** [review]
LGTM

**Pull request merged**